src/connector/build.rs

```diff
@@ -13,10 +13,11 @@
 // limitations under the License.
 
 fn main() {
-    let proto_dir = "./src/test_data/proto_recursive";
-
-    println!("cargo:rerun-if-changed={}", proto_dir);
+    // This function extracts WITH options from the Rust code and maintains
+    // them in a single source of truth `with_options.yaml`.
+    with_options_build::update_with_options_yaml();
```

Comment:
Why to put this in `build.rs`? Not strongly opposed to this, but feel a little strange. A bin might be more natural (like `generate-example-config`).

Reply:
Oh, we can do a `expect-test`, which can be used to both generate the file, and ensure it's up-to-date.

Reply:
Now the build script will rerun if there's any change in this crate. Have you checked the impact on the developer experience?

Reply:
LGTM!